### ISSUE TITLE
Model subclassing API ignores output layers names

### ISSUE THREAD

**Issue by dmirecki** [keras-team/keras#15118]
When defining multi-output model using subclassing API, names of the output layers are ignored. Despite naming the output layer, outputs from the model have names: `output_1`, `output_2`...

[Here](https://colab.research.google.com/drive/1cQzUH4WR3Ncu64bfV16siDM0be6PYf_p?usp=sharing) is the colab that reproduces the problem. Multiple loss functions have been used, specific loss has been assigned to specific output, so the losses were defined as dict. Unfortunately exception was thrown that the names of the losses do not correspond with the names of the model outputs.

The problem does not appear when functional API is used.

**System information**.
- TensorFlow version: 2.5.0

**Comment by jvishnuvardhan** [comment]
@dmirecki I changed your code little bit and it works without any issue. I only changed multi-outputs in the form of dictionary. Please take a look at the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/1c5a1f3b1b4991d431e4496574922b99/modelsubclassingoutputnames.ipynb).

Please verify and close the issue if this was resolved for you. Thanks!
